Dockerfile

```diff
@@ -62,6 +63,30 @@ RUN pip3 install --no-cache-dir -r ./docs/requirements.txt
 COPY . /code/listenbrainz
 
 
+#########################################################################
+# NOTE: The javascript files for production are compiled in this image. #
+#########################################################################
+FROM node:$NODE_VERSION as listenbrainz-frontend
+
+ARG NODE_VERSION
+
+LABEL org.label-schema.vcs-url="https://github.com/metabrainz/listenbrainz-server.git" \
+      org.label-schema.schema-version="1.0.0-rc1" \
+      org.label-schema.vendor="MetaBrainz Foundation" \
+      org.label-schema.name="ListenBrainz Static Builder" \
+      org.metabrainz.based-on-image="node:$NODE_VERSION"
+
+RUN mkdir /static
+WORKDIR /static
+
+COPY package.json package-lock.json /static/
+RUN npm install
+
+# Compile front-end (static) files
+COPY webpack.config.js babel.config.js .eslintrc.js .stylelintrc.js tsconfig.json ./listenbrainz/webserver/static /static/
+RUN npm run build:prod
+
```

Comment:
This is a separate image altogether. Only the content of /static folder is copied to the final image.

https://github.com/metabrainz/listenbrainz-server/blob/508b8db46bf85dec14966f2b793ea72ffc25ddfa/Dockerfile#L179

I guess it makes sense to do this if npm will not install the dev dependencies at all. As that may save time, if it removes as an extra step later then it'll just slow the build for no reason.

Reply:
We definitely need the dev dependencies in order to build (`npm run build:prod`)  so this suggestion would be indeed an extra step.
But I disagree with "for no reason": it would remove a lot of packages which we don't need once the JS is compiled, and which currently will get copied ot the final prod image (since it's all in listenbrainz-frontend's /static folder)